chains/orchestrator-relays/runtime/dancelight/src/tests/container_token_transfers.rs

```diff
@@ -848,3 +848,331 @@ fn native_container_can_process_message_returns_true_for_valid_message() {
+                channel_id,
+                agent_id,
+                para_id
+            ));
+
+            let token_addr = H160::repeat_byte(0x11);
+            let token_location = Location {
+                parents: 1,
+                interior: X2([
+                    GlobalConsensus(EthereumNetwork::get()),
+                    AccountKey20 {
+                        network: Some(EthereumNetwork::get()),
+                        key: token_addr.into(),
+                    },
+                ]
+                .into()),
+            };
+
+            let asset_id = 42u16;
+
+            assert_ok!(
+                pallet_foreign_asset_creator::Pallet::<Runtime>::create_foreign_asset(
+                    root_origin(),
+                    token_location.clone(),
+                    asset_id,
+                    AccountId::from(ALICE),
+                    true,
+                    1
+                )
+            );
+
+            let beneficiary = [5u8; 32];
+
+            let amount_to_transfer = 100_000_000;
+            let fee = 1_500_000_000_000_000;
+            let container_fee = 2_000_000_000_000_000;
+
+            let payload = VersionedXcmMessage::V1(MessageV1 {
+                chain_id: 1,
+                command: Command::SendToken {
+                    token: token_addr,
+                    destination: Destination::ForeignAccountId32 {
+                        para_id: container_para_id,
+                        id: beneficiary,
+                        fee: container_fee,
+                    },
+                    amount: amount_to_transfer,
+                    fee,
+                },
+            })
+            .encode();
+
+            let event = OutboundMessageAccepted {
+                channel_id: <[u8; 32]>::from(channel_id).into(),
+                nonce: 1,
+                message_id: Default::default(),
+                payload,
+            };
+
+            let message = EventProof {
+                event_log: Log {
+                    address:
+                        <Runtime as snowbridge_pallet_inbound_queue::Config>::GatewayAddress::get(),
+                    topics: event
+                        .encode_topics()
+                        .into_iter()
+                        .map(|w| H256::from(w.0 .0))
+                        .collect(),
+                    data: event.encode_data(),
+                },
+                proof: mock_snowbridge_message_proof(),
+            };
+
+            assert_ok!(EthereumInboundQueue::submit(relayer, message));
+
+            let sent = System::events().iter().any(|r| {
+                matches!(
+                    r.event,
+                    RuntimeEvent::XcmPallet(pallet_xcm::Event::Sent { .. })
+                )
+            });
+            assert!(sent, "XCM Sent event should be emitted!");
+        });
+}
+
+#[test]
+fn receive_container_foreign_tokens_from_eth_doesnt_error_if_error_sending_xcm() {
```

Comment:
Here, again, say that the reason why this is not working is because the para-head is not set